# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: netwerk/cookie/PCookieService.ipdl
# Commit: c8bfef0a3dd3
# Full Hash: c8bfef0a3dd356d4ed84089b179684aa42d687ff
# Author: Andrea Marchesini <amarchesini@mozilla.com>
# Date: 2020-04-06 21:47:50
# Regressor Bug: 1562868
# File Overlap Count: 1
# Description:
#   Bug 1562868 - Limit the amount of data shared between processes when document.cookie is set, r=smaug
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D68179
# ==============================================================================

diff -r 68475f12d4f1 -r c8bfef0a3dd3 netwerk/cookie/PCookieService.ipdl
--- a/netwerk/cookie/PCookieService.ipdl	Mon Apr 06 13:09:19 2020 +0000
+++ b/netwerk/cookie/PCookieService.ipdl	Mon Apr 06 14:03:59 2020 +0000
@@ -32,52 +32,11 @@
   manager PNecko;
 
 parent:
-
-  /*
-   * Set a cookie string.
-   *
-   * @param host
-   *        Same as the 'aURI' argument to nsICookieService.setCookieString.
-   * @param channelURI
-   *        The URI of the request.
-   * @param loadInfoArgs
-   *        The optional serialization of nsILoadInfo.
-   * @param isForeign
-   *        True if the the request is third party, for purposes of allowing
-   *        access to cookies. This should be obtained from
-   *        mozIThirdPartyUtil.isThirdPartyChannel. Third party requests may be
-   *        rejected depending on user preferences; if those checks are
-   *        disabled, this parameter is ignored.
-   * @param isThirdPartyTrackingResource
-   *        True if the request has been marked as 3rd party tracking.
-   * @param isThirdPartySocialTrackingResource
-   *        True if the request has been marked as 3rd party social tracking.
-   * @param firstPartyStorageAccessGranted
-   *        True if host has storage access granted. Note that the storage
-   *        access is automatically granted also if the channel is not marked as
-   *        tracking resource, or if it's not a 3rd party context.
-   * @param cookieString
-   *        Same as the 'aCookie' argument to nsICookieService.setCookieString.
-   * @param fromHttp
-   *        Whether the result is for an HTTP request header. This should be
-   *        true for nsICookieService.setCookieStringFromHttp calls, false
-   *        otherwise.
-   *
-   * @see nsICookieService.setCookieString
-   * @see nsICookieService.setCookieStringFromHttp
-   * @see mozIThirdPartyUtil.isThirdPartyChannel
-   */
-  nested(inside_cpow) async SetCookieString(URIParams host,
-                                            URIParams? channelURI,
-                                            LoadInfoArgs? loadInfoArgs,
-                                            bool isForeign,
-                                            bool isThirdPartyTrackingResource,
-                                            bool isThirdPartySocialTrackingResource,
-                                            bool firstPartyStorageAccessGranted,
-                                            uint32_t rejectedReason,
-                                            OriginAttributes aStoragePrincipalAttrs,
-                                            nsCString cookieString,
-                                            bool aFromHttp);
+  nested(inside_cpow) async SetCookies(nsCString baseDomain,
+                                       OriginAttributes attrs,
+                                       URIParams host,
+                                       bool fromHttp,
+                                       CookieStruct[] cookies);
 
   async PrepareCookieList(URIParams host,
                           bool isForeign,
